### PR TITLE
migration guide 0.12->0.13

### LINKED ISSUE #5129: Space View heuristics create SpaceViews for bounding boxes
Try the example `detect_and_track_objects`. You get a space-view for each bounding box, which is quite confusing (I thought at first these were images that were broken).

image

### PR DESCRIPTION
**Pull request by Wumpf**
### What

first stab at migration guide, creating a new subsection in the docs
**breaks link to 0.9 migration guide**

* Fixes #5129
   * the sample was already fixed by this pr writes down the changes that were necessary 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5136/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5136/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5136/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5136)
- [Docs preview](https://rerun.io/preview/70335ee724940e5d22f619b40da21b3d932102b2/docs) 
- [Examples preview](https://rerun.io/preview/70335ee724940e5d22f619b40da21b3d932102b2/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by Wumpf** [comment]
https://www.rerun.io/preview/b7a0797d3e8ecde350c0ad13b827d1bbbacda671/docs doesn't work. Guess that means I messed up the docs?

File: docs/content/reference/migration/migration-0-13.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: Migrating from 0.12
```
Comment (teh-cmc) [inline docs/content/reference/migration/migration-0-13.md:2]: I suggest making these very explicit, else people jumping version in the future might end up very confused...

This also applies to the existing 0.8->0.9 guide

```suggestion
title: Migrating from 0.12 to 0.13

File: docs/content/reference/migration/migration-0-13.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+order: 11
+---
+
+## [TimeSeriesScalar](../types/archetypes/time_series_scalar.md) scalar deprecated in favor of [Scalar](../types/archetypes/scalar.md) & [SeriesLine](../types/archetypes/series_line.md)/[SeriesPoint](../types/archetypes/series_point.md)
```
Comment (teh-cmc) [inline docs/content/reference/migration/migration-0-13.md:6]:
```suggestion
## [TimeSeriesScalar](../types/archetypes/time_series_scalar.md) deprecated in favor of [Scalar](../types/archetypes/scalar.md) & [SeriesLine](../types/archetypes/series_line.md)/[SeriesPoint](../types/archetypes/series_point.md)

File: docs/content/reference/migration/migration-0-13.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Previously, [TimeSeriesScalar](../types/archetypes/time_series_scalar.md) was used to define both
+data and styling of time series plots.
+Going forward, this is done separately: data is now logged via [Scalar](../types/archetypes/scalar.md).
+Styling for point/marker series via [SeriesPoint](../types/archetypes/series_point.md) and styling for
+lien series via [SeriesLine](../types/archetypes/series_line.md).
```
Comment (teh-cmc) [inline docs/content/reference/migration/migration-0-13.md:12]:
```suggestion
line series via [SeriesLine](../types/archetypes/series_line.md).

File: docs/content/reference/migration/migration-0-13.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+(Both series archetypes are typically logged as `timeless` but this is not a requirement and all properties may change over time!)
```
Comment (teh-cmc) [inline docs/content/reference/migration/migration-0-13.md:13]:
```suggestion
(Both styling archetypes are typically logged as `timeless` but this is not a requirement and all properties may change over time!)

File: docs/content/reference/migration/migration-0-13.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  * Example:
+    * Before: image @ `image/rgb`, rects at `image/annotation`
+    * After: image `image`, rects at `image/annotation`
+  * This happens because children of roots are no longer treated special for 2D views, but the viewer still
+    tries to bucket by image, putting images of the same size in the same view
+    * meaning the viewer no longer breaks up the root unless image-based bucketing implies it
+    * note that children of root are still special for 3D & time series views but this may change in the future
+      see [#4926](https://github.com/rerun-io/rerun/issues/4926)
+* [DisconnectedSpace](../types/archetypes/disconnected_space.md) now strictly applies only to 2D and 3D Space Views
+  * Internally, the heuristic reasons now about a 2D/3D topology which does not affect other types of views.
```
Comment (teh-cmc) [inline docs/content/reference/migration/migration-0-13.md:32]:
```suggestion
  * Internally, the heuristic now reasons about a 2D/3D topology which does not affect other types of views.

File: docs/content/reference/migration/migration-0-13.md
Comment (teh-cmc) [inline docs/content/reference/migration/migration-0-13.md:1]: You're missing a whole bunch of trailing double spaces for your new lines to be taken into account?

**Review by teh-cmc — APPROVED** [review]

File: docs/content/reference/migration/migration-0-13.md
Comment (Wumpf) [reply docs/content/reference/migration/migration-0-13.md:1]: ah, I'm fine without.
Also too hard to edit in an editor that wants to trim whitespace :D

**Review by teh-cmc — CHANGES_REQUESTED** [review]
We need to add a link to the guide in the deprecated attribute!

**Review by teh-cmc — APPROVED** [review]

**Pull request merged**